crates/rattler_cache/src/package_cache/mod.rs

```diff
@@ -98,17 +131,151 @@ impl From<Cancelled> for PackageCacheError {
     }
 }
 
+impl From<Cancelled> for PackageCacheLayerError {
+    fn from(_value: Cancelled) -> Self {
+        Self::Cancelled
+    }
+}
+
+impl From<PackageCacheLayerError> for PackageCacheError {
+    fn from(err: PackageCacheLayerError) -> Self {
+        // Convert the PackageCacheLayerError to a LayerError by boxing it
+        PackageCacheError::LayerError(Box::new(err))
+    }
+}
+
+impl PackageCacheLayer {
+    /// Determine if the layer is read-only in the filesystem
+    pub fn is_readonly(&self) -> bool {
+        self.path
+            .metadata()
+            .map(|m| m.permissions().readonly())
+            .unwrap_or(false)
+    }
+
+    /// Validate the package, and throw if invalid.
+    pub async fn validate_or_throw(
```

Comment:
This doesnt really "throw" anything. Maybe call `try_validate`?

Reply:
You're right, thanks!